FlowCrypt/Controllers/Compose/ComposeViewController.swift

```diff
@@ -211,57 +222,65 @@ extension ComposeViewController {
 
         showSpinner("sending_title".localized)
 
-        Promise<Bool> { [weak self] in
-            try await(self!.encryptAndSendMessage())
-        }.then(on: .main) { [weak self] sent in
-            if sent { // else it must have shown error to user
-                self?.handleSuccessfullySentMessage()
+        switch prepareMessage() {
+        case .failure(let error):
+            switch error {
+            case .internalError(let message):
+                showAlert(message: message)
+            case .missedSender:
+                showAlert(message: "compose_no_pub_sender".localized)
+            case .noPubKeyForRecipient:
+                // already handled
+                break
             }
-        }.catch(on: .main) { [weak self] error in
-            self?.showAlert(error: error, message: "compose_error".localized)
+        case .success(let data):
+            self.messageSender.sendMail(mime: data)
```

Comment:
Does `self.messageSender.sendMail` have to be in the result handling code? Could it be more similar to original code, where `encryptAndSendMessage` took care of sending as well?